src/WebJobs.Extensions.DurableTask/DurableOrchestrationClient.cs

```diff
@@ -68,6 +69,17 @@ public override HttpResponseMessage CreateCheckStatusResponse(HttpRequestMessage
                 functionType: FunctionType.Orchestrator,
                 isReplay: false);
 
+            if (waitUntilOrchestrationStarts)
+            {
+                DurableOrchestrationStatus status = await this.GetStatusAsync(instance.InstanceId);
+                Stopwatch stopwatch = Stopwatch.StartNew();
+                while ((status == null || status.RuntimeStatus == OrchestrationRuntimeStatus.Pending) && stopwatch.Elapsed < TimeSpan.FromSeconds(10))
```

Comment:
Lets change the wait time to 30 seconds and throw a timeout exception if it still hasn't started. Realistically, it should never take longer than 30 seconds to have an instance start and reach it's first checkpoint (which is when we would be able to see it's status). I worry that 10 seconds may be too short and 30 seconds is a common value to use for timeouts.

The reason I think a timeout would be better is because after 30 seconds something is obviously wrong and the user should be explicitly notified about this. The error message should say something like `$"Timeout expired while waiting for the new instance to start. This can happen if the task hub is overloaded or if the orchestration host failed to process the start message. Please check the orchestration logs to see whether an internal failure may have occurred. Instance ID: {instanceId}"`

Reply:
I think this change is very good because it simplifies the workflow and also follows best practices as you mentioned. Thank you!